pkg/utils/webhook/add.go

```diff
@@ -0,0 +1,158 @@
+			webhookPath := o.WebhookBasePath + elem.ResourceName
+			vwcWebhook := admissionregistrationv1.ValidatingWebhook{
+				Name:                    elem.ResourceName + o.WebhookNameSuffix,
+				SideEffects:             &noSideEffects,
+				FailurePolicy:           &failPolicy,
+				ObjectSelector:          &o.ObjectSelector,
+				AdmissionReviewVersions: []string{"v1"},
+				Rules:                   []admissionregistrationv1.RuleWithOperations{rule},
+				ClientConfig: admissionregistrationv1.WebhookClientConfig{
+					Service: &admissionregistrationv1.ServiceReference{
+						Namespace: o.ServiceNamespace,
+						Name:      o.ServiceName,
+						Path:      &webhookPath,
+						Port:      &o.ServicePort,
+					},
+					CABundle: o.CABundle,
+				},
+			}
+			vwcWebhooks = append(vwcWebhooks, vwcWebhook)
+		}
+
+		webhookLogger.Info("Creating ValidatingWebhookConfiguration", "name", o.WebhookConfigurationName, "kind", "ValidatingWebhookConfiguration")
+		_, err = ctrl.CreateOrUpdate(ctx, tmpClient, &vwc, func() error {
+			vwc.Webhooks = vwcWebhooks
+			return nil
+		})
+		if err != nil {
+			return fmt.Errorf("unable to create/update ValidatingWebhookConfiguration: %w", err)
+		}
+		webhookLogger.Info("ValidatingWebhookConfiguration created/updated", "name", o.WebhookConfigurationName, "kind", "ValidatingWebhookConfiguration")
+	} else {
+		webhookLogger.Info("Validation webhook disabled")
+		webhookLogger.Info("Removing ValidatingWebhookConfiguration, if it exists", "name", o.WebhookConfigurationName, "kind", "ValidatingWebhookConfiguration")
+		if err := tmpClient.Delete(ctx, &vwc); err != nil {
+			if apierrors.IsNotFound(err) {
+				webhookLogger.Info("ValidatingWebhookConfiguration not found", "name", o.WebhookConfigurationName, "kind", "ValidatingWebhookConfiguration")
+			} else {
+				webhookLogger.Error(err, "unable to delete validatingwebhookconfiguration", "name", o.WebhookConfigurationName, "kind", "ValidatingWebhookConfiguration")
+			}
+		} else {
+			webhookLogger.Info("ValidatingWebhookConfiguration deleted", "name", o.WebhookConfigurationName, "kind", "ValidatingWebhookConfiguration")
+		}
+	}
+
+	return nil
+}
+
+// RegisterWebhooks generates certificates and registers the webhooks to the manager
+// no-op if WebhookedResources in the given options is either nil or empty
+func RegisterWebhooks(ctx context.Context, mgr manager.Manager, o Options, webhookLogger logr.Logger) error {
+	if o.WebhookedResources == nil || len(o.WebhookedResources) == 0 {
+		return nil
+	}
+
+	// registering webhooks
+	for _, elem := range o.WebhookedResources {
+		val, err := ValidatorFromResourceType(elem.ResourceName)
+		if err != nil {
+			return fmt.Errorf("unable to register webhooks: %w", err)
+		}
+		if err := val.InjectClient(mgr.GetClient()); err != nil {
+			return fmt.Errorf("unable to register webhooks: %w", err)
+		}
+		rsLogger := webhookLogger.WithName(elem.ResourceName)
+		webhookPath := o.WebhookBasePath + elem.ResourceName
```

Comment:
```suggestion
		webhookPath := path.Join(o.WebhookBasePath, elem.ResourceName)
```
With that the function already takes care of concatination of 2 paths e.g. `"/some/path/" + "/some/other/path"`